### DISCUSSION TITLE
Are there any logs for filter behavior somewhere?

### DISCUSSION THREAD

**Discussion by DickenSerm** [Athou/commafeed#1464] [Q&A]
Hey, can I look somewhere which rule matched the RSS articles?
Every now and then something is marked as read for me even though according to the filter it shouldn't be marked at all.
I would love to be able to follow up on this.

**Comment by Athou** [comment]
Nothing is logged right now when an entry is marked as read because of a filtering expression, but I guess a debug line could be added.
Would something like "marking entry <entry_id> as read for user <user_id> because of a filtering expression on subscription <sub_id>" help?

**Reply by DickenSerm** [reply]
That wouldn't be bad, but then I still wouldn't know which filter was reacted to if you specified several, right?

**Reply by Athou** [reply]
Do you mean that if you have an expression like "title.contains('a') or title.contains('b')" you wouldn't know which was true? If that's the case I don't think I can do anything about that, I just pass the expression to the JEXL engine and I get a boolean back.

**Reply by DickenSerm** [reply]
Yes, that's what I mean. Let's take their example: "title.contains('a') or title.contains('b')"

Sometimes things are filtered that contain 'c' but neither 'a' nor 'b' 

Now I don't know if I'm setting the filters incorrectly or if it's something else.

Example:
title.contains('german') and not title.contains('english')

German Test Title = should be displayed
English Test Title = should not be displayed
German and English title = should not be displayed

Is it correct? 
Is there by any chance a site online where you can test something like this?

**Reply by Athou** [reply]
> Is it correct?

yep

> Is there by any chance a site online where you can test something like this?

nope

A page to test expressions could be added to CommaFeed I guess. I'd rather implement a real rule engine though, similar to what [InoReader is doing](https://www.inoreader.com/blog/wp-content/uploads/2019/03/2020-03-23_15-21-30-1.jpg).

**Reply by DickenSerm** [reply]
> yep

Ok, then that's right :D



> A page to test expressions could be added to CommaFeed I guess. I'd rather implement a real rule engine though, similar to what [InoReader is doing](https://www.inoreader.com/blog/wp-content/uploads/2019/03/2020-03-23_15-21-30-1.jpg).

That would of course be brilliant.
